src/DynamicDiscreteSamplers.jl

```diff
@@ -159,10 +159,53 @@ TODO
 # Trivial extensions:
 # push!, delete!
 
+Base.rand(rng::AbstractRNG, w::Weights, n::Integer) = _rand(rng, w.m, n)
 Base.rand(rng::AbstractRNG, w::Weights) = _rand(rng, w.m)
 Base.getindex(w::Weights, i::Int) = _getindex(w.m, i)
 Base.setindex!(w::Weights, v, i::Int) = (_setindex!(w.m, Float64(v), i); w)
 
+function _rand(rng::AbstractRNG, m::Memory{UInt64}, n::Integer)
+    n < 100 && return [_rand(rng, m) for _ in 1:n]
+    max_i = _convert(Int, m[2])
+    min_i = 5
+    k = 0
```

Comment:
This optimization is fairly self-contained. Let's put all of it in a file called `bulk_sampling.jl`

Reply:
All in the same file bulk_sampling.jl, not three files multinomial.jl, util.jl, and bulk_sampling.jl

Reply:
did it